src/components/accounts/two_factor/EnableTwoFactor.js

```diff
@@ -0,0 +1,185 @@
+import React, { useState, useEffect } from 'react';
+import { useSelector, useDispatch } from 'react-redux';
+import styled from 'styled-components';
+import { Translate } from 'react-localize-redux';
+import TwoFactorOption from './TwoFactorOption';
+import PhoneInput, { isValidPhoneNumber } from 'react-phone-number-input';
+import { validateEmail } from '../../../utils/account';
+import FormButton from '../../common/FormButton';
+import {
+    initTwoFactor,
+    verifyTwoFactor,
+    deployMultisig,
+    redirectToApp
+} from '../../../actions/account';
+import { useRecoveryMethods } from '../../../hooks/recoveryMethods';
+import EnterVerificationCode from '../EnterVerificationCode'
+import Container from '../../common/styled/Container.css'
+
+const StyledContainer = styled(Container)`
+
+    h4 {
+        margin-top: 40px;
+
+        span {
+            color: #FF585D;
+        }
+    }
+
+    button {
+        margin-top: 50px !important;
+        width: 100% !important;
+    }
+`
+
+export function EnableTwoFactor(props) {
+
+    const dispatch = useDispatch();
+    const account = useSelector(({ account }) => account);
+    const accountId = account.accountId;
+
+    const [initiated, setInitiated] = useState(false);
+    const [option, setOption] = useState('email');
+    const [email, setEmail] = useState('');
+    const [phoneNumber, setPhoneNumber] = useState('');
+    const recoveryMethods = useRecoveryMethods(accountId);
+    const loading = account.actionsPending.some(action => ['INIT_TWO_FACTOR', 'VERIFY_TWO_FACTOR', 'DEPLOY_MULTISIG'].includes(action))
+
+    const method = {
+        kind: `2fa-${option}`,
+        detail: option === 'email' ? email : phoneNumber
+    }
+
+    useEffect(() => { 
```

Comment:
not sure about few things:
- why use `useEffect` if this doesn't have actual side-effects? Like cannot this just be computed in `EnableTwoFactor` function directly (as with e.g. `loading`)?
- same point as above for updating `state`. Not sure why it's needed.

Reply:
@vgrichina I use `useEffect/componentDidMount` to pre-fill any email/phone inputs if available. And using state variable for both email/phone since I set those states with user input `onChange` too. Not sure what changes would make it more optimal, but please let me know if think there's a more optimal solution.